docs/metadata.rst

```diff
@@ -0,0 +1,81 @@
+.. _metadata:
+
+========
+Metadata
+========
+
+Samples are normally annotated with the use of ``descriptor`` and
+``descriptor_schema``. However in some cases the fields defined in
+``DescriptorSchema`` do not suffice and it comes handy to upload sample
+annotations in a table where each row holds information about some
+sample in collection. In general, there can be multiple rows referring
+to the same sample in the collection (for example one sample received
+two or more distinct treatments). In such cases one can upload this
+tables with the process `Metadata table`_. However, quite often there is
+exactly one-on-one mapping between rows in such table and samples in
+collection. In such case, please use the "unique" flavour of the above
+process, `Metadata table (one-to-one)`_.
+
+.. _Metadata table: https://resolwe-bio.readthedocs.io/en/latest/catalog-definitions.html#process-upload-metadata
+.. _Metadata table (one-to-one): https://resolwe-bio.readthedocs.io/en/latest/catalog-definitions.html#process-upload-metadata-unique
+
+Metadata in ReSDK is just a special kind of ``Data`` resource that
+simplifies retrieval of the above mentioned tables. In addition to all
+of the functionality of ``Data``, it also has two additional attributes:
+``df`` and ``unique``::
+
+    # The "df" attribute is pandas.DataFrame of the output named "table"
+    m.df
+    # Attribute "unique" is signalling if this is metadata is unique or not
+    gs.unique
+
+.. note::
+
+    Behind the scenes, ``df`` is not an attribute but rather a property.
+    SO it has getter and setter methods (``get_df`` and ``set_df``).
+    This comes handy if the default parsing logic does not suffice. In
+    such cases you can provide your own parser and keyword arguments for
+    it. Example::
+
+        import pandas
+        m.get_df(parser=pandas.read_csv, sep="\t", skiprows=[1, 2, 3])
+
+In the most common case, Metadata objects exist somewhere on Resolwe
+server and user just fetches them::
+
+    # Get one metadata by slug
+    m = res.metadata.get("my-slug")
+
+    # Filter metadata by some conditions, e.g. get all metadata
+    # from a given collection:
+    ms = res.metadata.filter(collection=<my-collection>):
```

Comment:
ms -> m

Reply:
Filter returns multile elements, so IMO, ms (as a plural of m) is OK.